infra_tooling_account/infra_tooling_account/infra_tooling_common_stack.py

```diff
@@ -67,11 +67,13 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
             topic_name=AWSNaming.SNSTopic(self, "internal-error"),
         )
 
-        # Notification SQS Queue
+        # Notification FIFO SQS Queue
         # TODO: confirm visibility timeout
         notification_queue = sqs.Queue(
             self,
             "salmonNotificationQueue",
+            content_based_deduplication=True,
```

Comment:
As I can see, to achieve messages deduplication with _message_group_id_ only, each _message_group_id_ should be unique: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/using-messagegroupid-property.html#avoding-processing-duplicates-in-multiple-producer-consumer-system

For FIFO queue it makes sense to have message grouping as in our case we don't really care about the ordering, but maybe it's also worth adding a unique **message deduplication ID** to the message to make sure there will be no duplicates regardless of _message_group_id_ ?

Reference: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/using-messagededuplicationid-property.html

Reply:
yes, right. To achieve messages deduplication we can either provide a unique message deduplication ID for each sent message or enable content-based deduplication 
https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/FIFO-queues-moving.html

At the moment I think that both options can work for us (I assume that we generate the messages with unique content)
 or do you already remember the case when the message content is not unique?